### PR TITLE
[sonic-cfggen] Add support to generate sample t1 config

### PR DESCRIPTION
**Pull request by taoyl-ms**
**- What I did**
Add support in `sonic-cfggen` to generate a sample LeafRouter configuration json file for any SKU. This is a preparation to remove sample minigraphs from platform folders and to use auto-generated sample config json files instead.

**- How to verify it**
`sonic-cfggen -H -k Mellanox-SN2700 --preset t1` will print a sample config_db.json file which is similar to the one generated from current default minigraph.

```
{
    "BGP_NEIGHBOR": {
        "10.0.0.1": {
            "rrclient": 0, 
            "name": "ARISTA01T2", 
            "local_addr": "10.0.0.0", 
            "nhopself": 0, 
            "holdtime": "180", 
            "asn": "65200", 
            "keepalive": "60"
        }, 
        "10.0.0.3": {
            "rrclient": 0, 
            "name": "ARISTA02T2", 
            "local_addr": "10.0.0.2", 
            "nhopself": 0, 
            "holdtime": "180", 
            "asn": "65200", 
            "keepalive": "60"
        }, 
        ...,
        "10.0.0.33": {
            "rrclient": 0, 
            "name": "ARISTA01T0", 
            "local_addr": "10.0.0.32", 
            "nhopself": 0, 
            "holdtime": "180", 
            "asn": "64001", 
            "keepalive": "60"
        }, 
        "10.0.0.35": {
            "rrclient": 0, 
            "name": "ARISTA02T0", 
            "local_addr": "10.0.0.34", 
            "nhopself": 0, 
            "holdtime": "180", 
            "asn": "64002", 
            "keepalive": "60"
        }, 
        ...
    }, 
    "DEVICE_METADATA": {
        "localhost": {
            "hwsku": "Mellanox-SN2700", 
            "platform": "x86_64-mlnx_msn2700-r0", 
            "mac": "7c:fe:90:5e:6a:80",
            "type": "LeafRouter", 
            "hostname": "sonic", 
            "bgp_asn": "65100"
        }
    }, 
    "DEVICE_NEIGHBOR": {}, 
    "LOOPBACK_INTERFACE": {
        "Loopback0|10.1.0.1/32": {}
    }, 
    "INTERFACE": {
        "Ethernet0|10.0.0.0/31": {}, 
        "Ethernet4|10.0.0.2/31": {}, 
        "Ethernet8|10.0.0.4/31": {}, 
        "Ethernet12|10.0.0.6/31": {}, 
        "Ethernet16|10.0.0.8/31": {}, 
        "Ethernet20|10.0.0.10/31": {}, 
        "Ethernet24|10.0.0.12/31": {}, 
        "Ethernet28|10.0.0.14/31": {}, 
        "Ethernet32|10.0.0.16/31": {}, 
        "Ethernet36|10.0.0.18/31": {}, 
        "Ethernet40|10.0.0.20/31": {}, 
        "Ethernet44|10.0.0.22/31": {}, 
        "Ethernet48|10.0.0.24/31": {}, 
        "Ethernet52|10.0.0.26/31": {}, 
        "Ethernet56|10.0.0.28/31": {}, 
        "Ethernet60|10.0.0.30/31": {}, 
        "Ethernet64|10.0.0.32/31": {}, 
        "Ethernet68|10.0.0.34/31": {}, 
        "Ethernet72|10.0.0.36/31": {}, 
        "Ethernet76|10.0.0.38/31": {}, 
        "Ethernet80|10.0.0.40/31": {}, 
        "Ethernet84|10.0.0.42/31": {}, 
        "Ethernet88|10.0.0.44/31": {}, 
        "Ethernet92|10.0.0.46/31": {}, 
        "Ethernet96|10.0.0.48/31": {}, 
        "Ethernet100|10.0.0.50/31": {}, 
        "Ethernet104|10.0.0.52/31": {}, 
        "Ethernet108|10.0.0.54/31": {}, 
        "Ethernet112|10.0.0.56/31": {}, 
        "Ethernet116|10.0.0.58/31": {}, 
        "Ethernet120|10.0.0.60/31": {}, 
        "Ethernet124|10.0.0.62/31": {}
    }, 
    "PORT": {
        "Ethernet0": {
            "alias": "Ethernet0", 
            "admin_status": "up", 
            "lanes": "0,1,2,3", 
            "mtu": "9100"
        }, 
        "Ethernet4": {
            "alias": "Ethernet4", 
            "admin_status": "up", 
            "lanes": "4,5,6,7", 
            "mtu": "9100"
        }, 
        "Ethernet8": {
            "alias": "Ethernet8", 
            "admin_status": "up", 
            "lanes": "8,9,10,11", 
            "mtu": "9100"
        }, 
        "Ethernet12": {
            "alias": "Ethernet12", 
            "admin_status": "up", 
            "lanes": "12,13,14,15", 
            "mtu": "9100"
        }, 
        ...
    }
}
```

### CODE REVIEW

File: src/sonic-config-engine/sonic-cfggen [hunk 1]
```diff
@@ -156,6 +157,7 @@ def main():
     group.add_argument("--var-json", help="print the value of a variable, in json format")
     group.add_argument("--write-to-db", help="write config into configdb", action='store_true')
     group.add_argument("--print-data", help="print all data", action='store_true')
+    group.add_argument("--preset", help="preset template name to generate a sample configuration")
```
Comment (lguohan) [inline src/sonic-config-engine/sonic-cfggen]: you add options here, so that user will know what presets you have.

**Comment by stcheng** [comment]
will there be any test cases to check the output?

File: src/sonic-config-engine/sonic-cfggen (hunk 1 above)
Comment (taoyl-ms) [reply src/sonic-config-engine/sonic-cfggen]: Added.

**Comment by taoyl-ms** [comment]
Added. Now user will be able to see a list of preset templates in the `-h` message.

**Pull request merged**